resources/git-rebase.md

```diff
@@ -0,0 +1,66 @@
+# Git rebase
+
+## What’s rebasing?
+
+Git rebase takes a branch you select and merges it on to the branch where you’re currently working. Then it takes all of your local commits and applies them on top.
+
+## You said it merges the branches. Why not use `git merge`?
+
+Git rebase solves the same issues as git merge, but git merge will make a merge commit every time you pull upstream changes. This can make your git history really noisy, really fast, making it difficult for you and your team to understand the history of your project.
+
```

Comment:
Do you want to get into the fact that a rebase moves your commits to the end of the git history (keeping it nice and clean) vs a merge which keeps the commits at the same point in history they were made but adds the merge commit at the end? (is that too complicated? the diagrams here helped me a lot when i was learning https://www.atlassian.com/git/tutorials/merging-vs-rebasing)

Reply:
yeah, can definitely expand here. Added the link as well - more diagrams are always better than less.